pkg/publish.go

```diff
@@ -97,3 +124,104 @@ func Publish(file string, rootID string, dryRun bool) (id string, err error) {
 	}
 	return c.ID, nil
 }
+
+func GetImageSrcsFromHTML(content string) []string {
+	var imageSrcs []string
+	z := html.NewTokenizer(strings.NewReader(content))
+	for {
+		tt := z.Next()
+		t := z.Token()
+		switch tt {
+		case html.ErrorToken:
+			return imageSrcs
+		case html.SelfClosingTagToken:
+			if t.Data == "img" {
+				for _, attr := range t.Attr {
+					if attr.Key == "src" {
+						imageSrcs = append(imageSrcs, attr.Val)
+					}
+				}
+			}
+		}
+	}
+	return imageSrcs
+}
+
+func GetImageSrcFiles(content string, file string) []string {
+	fileDir := filepath.Dir(file)
+	imageSrcs := GetImageSrcsFromHTML(content)
+
+	var imageSrcFiles []string
+
+	for _, imageSrc := range imageSrcs {
+		imageSrcPath := filepath.Join(fileDir, imageSrc)
+		if _, err := os.Stat(imageSrcPath); !os.IsNotExist(err) {
+			imageSrcFiles = append(imageSrcFiles, imageSrc)
+		}
```

Comment:
What can/should `dox` do when the image file does not exist?

Reply:
I cheat a lot here by simply checking if an image src exists. If it does, its an image file to be attached. If it doesn't, its likely to be a URL. As you say, there is also a chance that a user wants an image to be included, but it doesn't exist. 

I'll look into making this better. If we can sort out what looks like paths (or URLs), then we can send a warning if the file doesn't exist.